app/containers/ITunesProvider/TracksContainer/index.js

```diff
@@ -0,0 +1,140 @@
+/**
+ *
+ * TracksContainer
+ *
+ */
+
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import { createStructuredSelector } from 'reselect';
+import debounce from 'lodash/debounce';
+import get from 'lodash/get';
+import isEmpty from 'lodash/isEmpty';
+import { Input, Skeleton } from 'antd';
+import styled from 'styled-components';
+import { compose } from 'redux';
+import { useInjectSaga } from '@utils/injectSaga';
+import { makeSelectITunesProvider, selectTracksData, selectArtistName, selectTracksError } from '../selectors';
+import For from '@components/For';
+import { iTunesProviderCreators } from '../reducer';
+import saga from '../saga';
+import TrackCard from '@components/TrackCard';
+
+const { Search } = Input;
+
+const Container = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    max-width: ${props => props.maxWidth}em;
+    width: 100%;
+    margin: 0 auto;
+  }
+`;
+export function TracksContainer({
+  maxWidth,
+  dispatchFetchTracks,
+  dispatchClearTracks,
+  artistName,
+  tracks,
+  tracksError
+}) {
+  useInjectSaga({ key: 'iTunesProvider', saga });
+  const [loading, setLoading] = useState(false);
```

Comment:
do we need another loading state variable? 

There is a "loader" that we can read from the redux store yes?

Reply:
@mashrukh-islam